src/models/tshirt/tshirt_c.cpp

```diff
@@ -0,0 +1,809 @@
+ * @param num_timesteps The number of time steps, which PROBABLY should always be 1
+ * @param giuh_ordinates A pointer to the head of an array of GIUH ordinate values
+ * @param num_giuh_ordinates The number of elements in the array pointed to by ``giuh_ordinates``
+ * @param runoff_queue_m_per_timestep Pointer to head of array for GIUH convolution queue, of size ``num_giuh_ordinates`` + 1
+ * @param field_capacity_atm_press_fraction Fraction of atmospheric pressure for field capacity suction pressure (also alpha_fc)
+ * @param water_table_slope Assumed near channel water table slope lateral flow parameter
+ * @param Schaake_adjusted_magic_constant_by_soil_type Schaake magic constant for soil type
+ * @param lateral_flow_linear_reservoir_constant Later flow reservoir constant (Klf)
+ * @param K_nash Nash reservoir flow constant
+ * @param num_lateral_flow_nash_reservoirs The number of reservoirs in the cascade for the lateral flow
+ * @param nash_storage Pointer to head of array of already-set Nash Cascade reservoir storage values
+ * @param yes_aorc Whether
+ * @param aorc_data Pointer to head of array of per-time-step AORC data structs, when yes_aorc is TRUE (array of size num_timesteps)
+ * @param rain_rate Pointer to head of array of per-time-step simple rain rate data (in mm/h), when yes_aorc is FALSE (array of size num_timesteps)
+ * @param num_added_fluxes Reference for number of fluxes added to 'fluxes' parameter array; initialized to 0.
+ * @param fluxes Pointer to head of array of array per-time-step resulting flux values (array of size num_timesteps)
+ * @return
+ */
+extern int run(NWM_soil_parameters& NWM_soil_params,
+               conceptual_reservoir& gw_reservoir,
+               conceptual_reservoir& soil_reservoir,
+               int num_timesteps,
+               double* giuh_ordinates,
+               int num_giuh_ordinates,
+               double *runoff_queue_m_per_timestep,
+               double field_capacity_atm_press_fraction,
+               double water_table_slope,
+               double Schaake_adjusted_magic_constant_by_soil_type,
+               double lateral_flow_linear_reservoir_constant,
+               double K_nash,
+               int num_lateral_flow_nash_reservoirs,
+               double* nash_storage,
+               int yes_aorc,
+               aorc_forcing_data* aorc_data,
+               double* rain_rate,
+               int& num_added_fluxes,
+               tshirt_c_result_fluxes* fluxes)
+{
+    // Do this to be safe ...
+    num_added_fluxes = 0;
+
+    /* Commenting out, since this will need to return values rather than write results to a file.
+    FILE *in_fptr;
+    FILE *out_fptr;
+    FILE *out_debug_fptr;
+    */
+
+    // local variables
+    //-------------------------------------------------------------------
+    int i;
+    int tstep;
+    double upper_lim;
+    double lower_lim;
+    double diff;
+    char theString[513];   // dangerously hard coded string size... TODO fixme.
```

Comment:
looks like this can go away, was used in the original implementation to read the AORC csv lines from file, this data is now passed into this function, so this shouldn't be needed.